components/tidb_query_vec_executors/src/index_scan_executor.rs

```diff
@@ -52,23 +53,45 @@ impl<S: Storage> BatchIndexScanExecutor<S> {
         // Note 3: Currently TiDB may send multiple PK handles to TiKV (but only the last one is
         // real). We accept this kind of request for compatibility considerations, but will be
         // forbidden soon.
-        let decode_handle = columns_info.last().map_or(false, |ci| ci.get_pk_handle());
+        use DecodeHandleStrategy::*;
+
+        let is_int_handle = columns_info.last().map_or(false, |ci| ci.get_pk_handle());
+        let is_common_handle = primary_column_ids_len > 0;
+        let (decode_strategy, handle_column_cnt) = match (is_int_handle, is_common_handle) {
+            (false, false) => (NoDecode, 0),
+            (false, true) => (DecodeCommonHandle, primary_column_ids_len),
+            (true, false) => (DecodeIntHandle, 1),
+            // TiDB may accidentally push down both int handle or common handle.
+            // However, we still try to decode int handle.
+            _ => {
+                return Err(other_err!(
+                    "Both int handle and common handle are push downed"
+                ));
+            }
+        };
+
+        if handle_column_cnt > columns_info.len() {
```

Comment:
```suggestion
        if handle_column_cnt >= columns_info.len() {
```

Reply:
It may have a chance that all the columns are contained by the common handle.

Reply:
I think this is not possible. The index key must contain at least 1 column which is not a handle.